### PR TITLE
Create store, write tests

### PR DESCRIPTION
**Pull request by nvkuznetsova**

### CODE REVIEW

File: src/routes.tsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+import { ROUTES } from './constants/routes';
+import { ProductsPageContainer } from './pages/products';
+
+export const routes = (
+  <BrowserRouter>
+    <Switch>
+      <Redirect exact from="/" to={ROUTES.products} />
+      {/* Component does not render */}
+      <Route exact path={ROUTES.products} component={ProductsPageContainer} />
```
Comment (nvkuznetsova) [inline src/routes.tsx]: Сomponent stopped rendering after adding routing (PRODUCTS_LOAD_START action is not dispatched). Without routing it is ok.

File: src/api/api.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import axios from 'axios';
+const baseURL = 'http://localhost:4000';
+
+export const getProducts = () => {
+  return axios.get(`${baseURL}/products`, {
```
Comment (ArtemX9) [inline src/api/api.ts]: better to extract initialization of axioms into api.ts and declare routes according to their domain, i.e. 

/api/api.ts
/api/products.ts

File: src/index.tsx [hunk 3]
```diff
@@ -1,12 +1,16 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+
+
```
Comment (ArtemX9) [inline src/index.tsx]: watch code style

File: src/index.tsx [hunk 4]
```diff
@@ -1,12 +1,16 @@
...
+import { Provider } from 'react-redux';
```
Comment (ArtemX9) [inline src/index.tsx:3]: usually libraries are declared on top

File: src/index.tsx [hunk 5]
```diff
@@ -1,12 +1,16 @@
...
 import './index.css';
```
Comment (ArtemX9) [inline src/index.tsx]: styles are declared below all imports

File: src/pages/products/ProductsPage.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      products,
+      isLoading,
+      error,
+    } = this.props;
+ 
+    return (
+    <div className="d-flex flex-wrap justify-content-evenly">
+      {isLoading && <span>Loading</span>}
+      {!isLoading && products.map((product, index) => {
+      return <div className="col-md-3 align-self-center" key={index}>{product.name}</div>
```
Comment (ArtemX9) [inline src/pages/products/ProductsPage.tsx]: watch codestyle

File: src/pages/products/ProductsPage.tsx [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+      })}
+      {!isLoading && !isEmpty(error) && <div>{get(error, 'message', 'error')}</div>}
```
Comment (ArtemX9) [inline src/pages/products/ProductsPage.tsx]: very hard to read this line, please extract into separate render function

File: src/routes.tsx [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { BrowserRouter, Route, Switch, Redirect } from 'react-router-dom';
+
+import { ROUTES } from './constants/routes';
+import { ProductsPageContainer } from './pages/products';
+
+export const routes = (
```
Comment (ArtemX9) [inline src/routes.tsx]: why not make it a default export?

File: src/store/products/__test__/products-reducer.test.ts [hunk 9]
```diff
@@ -0,0 +1,85 @@
+import { ProductsActionTypes, ProductsActions } from '../actions';
+
+import { productsReducer } from '../products-reducer';
+
+describe('ProductsReducer', () => {
+  it('should return new state on PRODUCTS_LOAD_START', () => {
```
Comment (ArtemX9) [inline src/store/products/__test__/products-reducer.test.ts]: better to break long tests into chunks:
```js
describe('ProductsReducer', () => {
  describe('PRODUCTS_LOAD_START', () => {
    it(...)
});
  describe('PRODUCTS_LOAD_SUCCESS', () => {
    it(...)
});
  describe('PRODUCTS_LOAD_ERROR', () => {
    it(...)
});

File: src/store/products/actions/__test__/producst-actions.test.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+import {
+  ProductsActionTypes,
+  productsLoadStart,
+  productsLoadSuccess,
+  productsLoadError,
+ } from '..';
+import { Product } from '../../../../domain/Product';
```
Comment (ArtemX9) [inline src/store/products/actions/__test__/producst-actions.test.ts]: try using absolute routes: 'src/domain/Product', like this

File: src/api/api.ts (hunk 2 above)
Comment (nvkuznetsova) [reply src/api/api.ts]: Fixed

File: src/index.tsx (hunk 3 above)
Comment (nvkuznetsova) [reply src/index.tsx]: Fixed

File: src/index.tsx (hunk 4 above)
Comment (nvkuznetsova) [reply src/index.tsx:3]: Fixed

File: src/pages/products/ProductsPage.tsx (hunk 6 above)
Comment (nvkuznetsova) [reply src/pages/products/ProductsPage.tsx]: Fixed

File: src/pages/products/ProductsPage.tsx (hunk 7 above)
Comment (nvkuznetsova) [reply src/pages/products/ProductsPage.tsx]: Done

File: src/routes.tsx (hunk 8 above)
Comment (nvkuznetsova) [reply src/routes.tsx]: Done

File: src/store/products/__test__/products-reducer.test.ts (hunk 9 above)
Comment (nvkuznetsova) [reply src/store/products/__test__/products-reducer.test.ts]: Done

File: src/store/products/actions/__test__/producst-actions.test.ts (hunk 10 above)
Comment (nvkuznetsova) [reply src/store/products/actions/__test__/producst-actions.test.ts]: Done

**Pull request merged**